drbert/evaluation/eval.py

```diff
@@ -0,0 +1,225 @@
+import logging
+
+import torch
+from sklearn.metrics import precision_recall_fscore_support
+from tqdm import tqdm
+
+from ..constants import OUTSIDE
+from ..constants import SEQUENCE_CLASSIFICATION_TASKS
+from ..constants import TASKS
+from ..training.util import generate_inputs
+from .util import precision_recall_f1_support_sequence_labelling
+from .util import print_evaluation
+from .util import reverse_dict
+
+logger = logging.getLogger(__name__)
+
+
+def evaluate(tasks, model, tokenizer, partitions=None):
+    """The interface for evaluating DrBERT.
+
+    This function handles evaluation of DrBERT (`model`) for some tasks (`tasks`).
+
+    Args:
+        tasks (list): A list of dictionaries containing information about each task to evaluate.
+        model (model.DrBERT): The DrBERT model to evaluate.
+        tokenizer (PretrainedTokenizer): A transformers tokenizer object.
+        partitions (list, optional): A list of partition names to evaluate. If None, each partition
+            in task['partitions']` is evaluated for each task in tasks. Defaults to None.
+
+    Returns:
+        dict: A dictionary (keyed by task names, `task['name']`) of dictionaries (keyed by
+            partitions).
+    """
+    model.eval()
+
+    results = {}
+
+    for task in tasks:
+        results[task['name']] = {}
+        partitions = task['partitions'] if partitions is None else partitions
+
+        # task_step = {task['name']: 0 for task in tasks}
+        # eval_loss = {task['name']: 0.0 for task in tasks}
+
+        for partition in partitions:
+            logger.info(f"***** Running {task['name']} evaluation on {partition} *****")
+            logger.info(f"  Num examples = {len(task['iterators'][partition])}")
+            logger.info(f"  Batch size = {task['iterators'][partition].batch_size}")
+
+            y_true, y_pred = [], []
+            if task['task'] == 'sequence_labelling':
+                orig_tok_mask = []
+
+            eval_iterator = tqdm(
+                task['iterators'][partition], desc="Evaluating", unit='batch', dynamic_ncols=True
+            )
+
+            for batch in eval_iterator:
+                inputs = generate_inputs(task['name'], task['task'], batch, tokenizer)
+
+                outputs = model(**inputs)
+                loss, logits = outputs[:2]
+
+                y_true.extend(batch.label.tolist())
+                y_pred.extend(torch.argmax(logits, dim=-1).tolist())
+
+                if task['task'] == 'sequence_labelling':
+                    orig_tok_mask.extend(batch.mask.tolist())
+
+                del batch, inputs, outputs, loss, logits
+
+            idx_to_label = \
+                reverse_dict(task['iterators'][partition].dataset.fields['label'].vocab.stoi)
+
+            if task['task'] == 'sequence_labelling':
+                scores = evaluate_sequence_labelling(y_true, y_pred, idx_to_label)
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [No value for argument 'orig_tok_mask' in function call](https://app.codacy.com/app/JohnGiorgi/drbert/pullRequest?prid=4348433)